Add HAN pullup config and mosquito board

dbeinder opened the pull request:
As mentioned in https://github.com/UtilitechAS/amsreader-firmware/issues/439

- ESP32-C3: fix UART0 pin (GPIO20) and limits the drop down to actually available GPIOs
- Adds a config option to control the internal pull up on the HAN RX pin
- Adds my [µHAN mosquito](https://github.com/UtilitechAS/amsreader-firmware/discussions/480) board as a custom hardware

gskjold reviewed (commented):
Nicely done, good to know that others also are able to navigate my code :) Minor changes suggested

dbeinder commented:
Oh, I thought Github supports updating a PR using amend/force-push and can show the changes since the already reviewed version, apparently that's only in Gitlab, my apologies.

I've reverted the config version comments and removed the baud defaults. Since my HW has an inverting level-shifter, I needed this additional change to get auto-baud working:
https://github.com/UtilitechAS/amsreader-firmware/blob/35d47902c63d4e3fcad4a5df02b2338f848ce0c3/src/AmsToMqttBridge.ino#L648-L654

There is a wrinkle though: my meter sends 2400/8E1, but auto-baud detects it as 8N1, which works fine, surprisingly. Probably the combination of the ESP serial peripheral being able to just ignore the parity bit, and the relatively slow update rate of my meter (15sec) makes that happen?

gskjold commented:
> There is a wrinkle though: my meter sends 2400/8E1, but auto-baud detects it as 8N1, which works fine, surprisingly. Probably the combination of the ESP serial peripheral being able to just ignore the parity bit, and the relatively slow update rate of my meter (15sec) makes that happen? It's not ideal, but I can live with this behavior, maybe I can find a way to improve this later.

Yeah, I have seen this before. Have not yet found a good solution for this... Let me know if you ever solve it :)
